specs/projectreuniondeployment/projectreuniondeployment.md

```diff
@@ -0,0 +1,138 @@
+require operating system updates in order to deploy. While this may be disruptive to the user to
+have to resolve this issue, it can be mitigated by usage of AppInstaller and Store updates to keep
+the Project Reunion packages up-to-date to minimize the likelihood and frequency of the user
+requiring direct action. The ideal situation is that the user needs to do this at most once, and
+auto-update mechanisms take care of future updates. It can be completely mitigated on newer
+operating systems that can have the updates and Project Reunion packages already pre-installed.
+Still, the developer of programs and app packages can never assume that the version of Project
+Reunion they require is already available, so this feature will provide an API to check that
+availability and provide links for resolving any deployment issues. This API could be expanded
+in the future to wrap deployment behavior for software capable of deploying packages and/or OS
+updates to reduce the end-user friction and minimize potential for developers to lose customers
+due to dropoff during the acquistion of Project Reunion.
+
+# Description
+
+When a Project Reunion-consuming package or app is deployed, it is expected to only have the App
+package or installer itself, and the Project Reunion main and framework packages may not be
+present, or may not be at the desired version. This API enables a developer to detect when the
+required Project Reunion main and/or framework packages are present, and to direct the user to the
+appropriate resources to acquire or update Project Reunion if it is absent or requires updates.
+
+Since this API is part of the Project Reunion framework, the framework itself must be present.
+For MSIX packages, the framework must be listed as a dependency in the application manifest and
+will not be deployable without a min version being present. For Win32 programs and installers the
+Project Reunion bootstrapper will try to load the framework and error out with a similar
+instruction on how to resolve the missing dependencies Project Reunion.
+
+Developers must not assume that the Project Reunion main package is present, or that the version
+present is the minimum version required to support the feature set they expect. Instead, this API
+should be used to validate any Project Reunion requirements and direct the user to resolve any
+missing dependencies.
+
+The developer scenarios for this API is to answer the questions:
+1. Does this system have the min version of Project Reunion I require?
+2. What do I do to resolve this if it doesn't?
+
+The API addresses #1 by a simple query of the version they seek and a possible check for OS
+updates. #2 has two possible answers. First is direct resolution by the developer by acquiring and
+installing the packages if possible. Some programs will not have permission or motivation to do
+this, so the second answer is providing a link up to the user where the user can take action to
```

Comment:
By "user" are you referring to someone using Windows, or a developer? Popping up install messages to the user sounds icky. When and why would this happen?